frontend/packages/console-plugin-sdk/src/store.ts

```diff
@@ -290,17 +295,29 @@ type LoadedDynamicPlugin = {
   enabled: boolean;
 };
 
+type FailedDynamicPlugin = {
+  errorMessage: string;
+  errorCause?: unknown;
+};
+
 export type LoadedDynamicPluginInfo = {
   status: 'Loaded';
   pluginID: string;
   metadata: DynamicPluginMetadata;
   enabled: boolean;
 };
 
-export type NotLoadedDynamicPluginInfo = {
-  status: 'Pending' | 'Failed';
-  pluginName: string;
-};
+export type NotLoadedDynamicPluginInfo =
+  | {
+      status: 'Pending';
```

Comment:
nit: We should be creating an enum for these statuses.

Reply:
I think it's fine to use type unions here, TypeScript will infer the specific type based on properties that discriminate between different union members.

For example, if an object typed as `NotLoadedDynamicPluginInfo` has `status: 'Failed'` then it must meet the structural requirement of a union member
```
{
  status: 'Failed';
  pluginName: string;
  errorMessage: string;
  errorCause?: unknown;
}
```

Creating an enum with members from each union member is possible, but not technically necessary unless we want to give these members explicit names. In this specific case I'd prefer to just use a union :smiley:

Reply:
Edit: I misunderstood the review comment :disappointed: 

Yes, we can create either a string union or an enum for the plugin info `status` property.